CacheStorageManager: don't create stray salt file in CWD

mxschmitt opened the pull request:
#### 8dfdc492a47b625177409bbc2aa49a7a24f1398c
<pre>
CacheStorageManager: don&apos;t create stray salt file in CWD
<a href="https://bugs.webkit.org/show_bug.cgi?id=254663">https://bugs.webkit.org/show_bug.cgi?id=254663</a>

Reviewed by Sihui Liu.

When there is no |cacheStoragePath| given, it created a file named &apos;salt
in the CWD. This started happening after 5d9c260 and regressed there.

* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
(WebKit::CacheStorageManager::readOrMakeSalt const):
(WebKit::CacheStorageManager::CacheStorageManager):
(WebKit::CacheStorageManager::saltFilePath const): Deleted.
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.h:

Canonical link: <a href="https://commits.webkit.org/262291@main">https://commits.webkit.org/262291@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/56f98cccd239ba98ddd48c9b2af251723758e47b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1114 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1144 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1183 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/1776 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/983 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1194 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1224 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1177 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1122 "Passed tests") | [⏳ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/iOS-16-Simulator-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1032 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/1648 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1029 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1019 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1016 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1059 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2108 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1061 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/989 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1002 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1039 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/284 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1082 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
```diff
@@ -223,15 +223,17 @@ void CacheStorageManager::makeDirty()
 }
 
-String CacheStorageManager::saltFilePath() const
+FileSystem::Salt CacheStorageManager::readOrMakeSalt() const
```
This does not have to be a class method. Let's make it:
`static Filesystem::Salt readOrMakeSalt(const String& saltDirectory)` and `readOrMakeSalt(m_path)` below

mxschmitt replied:
Thanks, addressed.

szewai reviewed (approved):
LGTM

mxschmitt committed:
CacheStorageManager: don't create stray salt file in CWD
https://bugs.webkit.org/show_bug.cgi?id=254663

Reviewed by Sihui Liu.

When there is no |cacheStoragePath| given, it created a file named 'salt
in the CWD. This started happening after 5d9c260 and regressed there.

* Source/WebKit/NetworkProcess/storage/CacheStorageManager.cpp:
(WebKit::CacheStorageManager::readOrMakeSalt const):
(WebKit::CacheStorageManager::CacheStorageManager):
(WebKit::CacheStorageManager::saltFilePath const): Deleted.
* Source/WebKit/NetworkProcess/storage/CacheStorageManager.h:

Canonical link: https://commits.webkit.org/262291@main
